src/gui/accountsettings.cpp

```diff
@@ -243,82 +243,116 @@ void AccountSettings::slotCustomContextMenuRequested(const QPoint &pos)
     };
+
     QTreeView *tv = ui->_folderList;
     QModelIndex index = tv->indexAt(pos);
     if (!index.isValid()) {
         return;
-    } else if (!_model->data(index, FolderStatusDelegate::IsReady).toBool()) {
+    }
+
+    auto classification = _model->classify(index);
+    if (classification != FolderStatusModel::RootFolder && classification != FolderStatusModel::SubFolder) {
+        return;
+    }
+
+
+    // Only allow removal if the item isn't in "ready" state.
+    if (classification == FolderStatusModel::RootFolder && !_model->data(index, FolderStatusDelegate::IsReady).toBool()) {
         QMenu *menu = new QMenu(tv);
         menu->setAttribute(Qt::WA_DeleteOnClose);
         removeFolderAction(menu);
         menu->popup(QCursor::pos());
         return;
     }
 
-    if (_model->classify(index) == FolderStatusModel::SubFolder) {
-        QMenu *menu = new QMenu(tv);
-        menu->setAttribute(Qt::WA_DeleteOnClose);
+    QMenu *menu = new QMenu(tv);
+    menu->setAttribute(Qt::WA_DeleteOnClose);
 
-        QAction *ac = menu->addAction(tr("Open local folder"));
-        connect(ac, &QAction::triggered, this, &AccountSettings::slotOpenCurrentLocalSubFolder);
+    // Add an action to open the folder in the system's file browser:
 
+    QUrl folderUrl;
+    if (classification == FolderStatusModel::SubFolder) {
         QString fileName = _model->data(index, FolderStatusDelegate::FolderPathRole).toString();
-        if (!QFile::exists(fileName)) {
+        folderUrl = QUrl::fromLocalFile(fileName);
+    } else {
+        // the root folder
+        auto alias = _model->data(index, FolderStatusDelegate::FolderAliasRole).toString();
+        if (Folder *f = FolderMan::instance()->folder(alias)) {
+            folderUrl = QUrl::fromLocalFile(f->path());
+        }
+    }
+
+    if (!folderUrl.isEmpty()) {
+        QString openLocallyLabel;
+        if (Utility::isWindows()) {
+            openLocallyLabel = tr("Show in Explorer");
+        } else if (Utility::isMac()) {
+            openLocallyLabel = tr("Show in Finder");
+        } else {
+            openLocallyLabel = tr("Show if file manager");
+        }
+
+        QAction *ac = menu->addAction(openLocallyLabel);
+        connect(ac, &QAction::triggered, [folderUrl]() {
+            qCInfo(lcAccountSettings) << "Opening local folder" << folderUrl;
+            if (!QDesktopServices::openUrl(folderUrl)) {
+                qCWarning(lcAccountSettings) << "QDesktopServices::openUrl failed for" << folderUrl;
+            }
+        });
+
+        if (!QFile::exists(folderUrl.toLocalFile())) {
             ac->setEnabled(false);
         }
+    }
+
+    // Add an action to open the folder on the server in a webbrowser:
 
-        ac = menu->addAction(tr("Open folder in browser"));
-        auto info = _model->infoForIndex(index);
-        OC_ASSERT(info);
+    if (auto info = _model->infoForIndex(index)) {
+        QAction *ac = menu->addAction(tr("Open Remote Folder in Browser"));
```

Comment:
see `addAction(openLocallyLabel, this, []{`

Reply:
Open in web browser maybe?

Reply:
Hmm, I wanted to make it clear that it's the *remote* folder that's being opened. But maybe that's too obvious. What do you think?

Reply:
Yes but what is a remote folder :D
and we should align it with the others.

![image](https://user-images.githubusercontent.com/200626/152531049-d895939b-5bf4-4bce-9604-065f90d21e6a.png)